Tests/OpenPassTests/DeviceAuthorizationFlowTests.swift

```diff
@@ -0,0 +1,266 @@
+//
+// MIT License
+//
+// Copyright (c) 2024 The Trade Desk (https://www.thetradedesk.com/)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+@testable import OpenPass
+import XCTest
+
+class ImmediateClock: Clock {
+    func sleep(nanoseconds duration: UInt64) async throws {
+        // Proceed immediately
+        return
+    }
+}
+
+extension DeviceAuthorizationFlow {
+    static func test(
+        openPassClient: OpenPassClient = .test,
+        tokenValidator: IDTokenValidation = IDTokenValidationStub.valid,
+        tokensObserver: @escaping ((OpenPassTokens) async -> Void) = { _ in },
+        dateGenerator: DateGenerator = .init({ Date(timeIntervalSince1970: 1600000000) }),
+        clock: Clock = ImmediateClock()
+    ) -> DeviceAuthorizationFlow {
+        DeviceAuthorizationFlow(
+            openPassClient: openPassClient,
+            tokenValidator: tokenValidator,
+            tokensObserver: tokensObserver,
+            dateGenerator: dateGenerator,
+            clock: clock
+        )
+    }
+}
+
+final class DeviceAuthorizationFlowTests: XCTestCase {
+    var flow: DeviceAuthorizationFlow!
+
+    @MainActor
+    override func setUp() {
+        flow = DeviceAuthorizationFlow.test()
+    }
+
+    // MARK: - Device Code
+
+    @MainActor
+    func testDeviceCodeAvailableState() async throws {
+        try HTTPStub.shared.stub(fixtures: [
+            "/v1/api/authorize-device" : ("authorize-device-200", 200),
+            "/.well-known/jwks": ("jwks", 200),
+        ])
+
+        let deviceCode = try await flow.fetchDeviceCode()
+        XCTAssertEqual(
+            deviceCode,
+            DeviceCode(
+                userCode: "T4UGZ6RK",
+                verificationUri: "https://auth.myopenpass.com/device",
+                verificationUriComplete: "https://auth.myopenpass.com/device?user_code=T4UGZ6RK",
+                expiresAt: Date(timeIntervalSince1970: 1600000500),
```

Comment:
Nit: Should `1600000000` be stored somewhere and shared between the `test` function and this?